openedx/core/djangoapps/credit/models.py

```diff
@@ -392,6 +396,13 @@ def get_course_requirement(cls, course_key, namespace, name):
             return None
 
 
+@receiver(models.signals.post_save, sender=CreditRequirement)
+@receiver(models.signals.post_delete, sender=CreditRequirement)
+def invalidate_credit_requirement_cache(sender, **kwargs):   # pylint: disable=unused-argument
+    """Invalidate the cache of credit requirements. """
+    RequestCache.clear_request_cache(name=CreditRequirement.CACHE_NAMESPACE)
```

Comment:
It's unclear to me whether that's really needed.  Since the cache is request-specific (and not in-memory across requests), I believe this never happens today.  That is, is there a request that both reads and writes this data across multiple courses?   If this cache wasn't request-specific, then I can see the need to make it more complex.  Thoughts?